File: examples/node-worker/worker.js
import { Worker } from "@feather/sdk";

const worker = new Worker({
  address: process.env.FEATHER_ADDRESS ?? "localhost:50051",
  workerId: process.env.WORKER_ID ?? `node-${process.pid}`,
  queues: ["default"],
});

worker.use(async (ctx, next) => {
  console.log("job", ctx.id, ctx.name);
  await next();
});

worker.task("echo", async (ctx) => {
  const body = JSON.parse(ctx.payload.toString() || "{}");
  console.log("echo payload", body);
});

console.log("worker starting...");
await worker.start();

File: examples/python-worker/worker.py
#!/usr/bin/env python3
"""Python worker example."""

import asyncio
import json
import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "..", "packages", "sdk-python"))

from getfeather import Worker


async def echo(ctx) -> None:
    body = json.loads(ctx.payload.decode() or "{}")
    print("echo payload", body)


async def main() -> None:
    worker = Worker()
    worker.task("echo", echo)
    print("worker starting...")
    await worker.start()


if __name__ == "__main__":
    asyncio.run(main())
